dataclients/kubernetes/ingress.go

```diff
@@ -421,3 +449,25 @@ func (ing *ingress) convert(state *clusterState, df defaultFilters) ([]*eskip.Ro
 
 	return routes, nil
 }
+
+func generateTLSCertFromSecret(secret *secret) (*tls.Certificate, error) {
+	if secret.Data[tlsSecretDataCrt] == "" || secret.Data[tlsSecretDataKey] == "" {
+		return nil, fmt.Errorf("secret must contain %s and %s in data field", tlsSecretDataCrt, tlsSecretDataKey)
+	}
+	crt, err := b64.StdEncoding.DecodeString(secret.Data[tlsSecretDataCrt])
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode secret data %s", tlsSecretDataCrt)
+	}
+	key, err := b64.StdEncoding.DecodeString(secret.Data[tlsSecretDataKey])
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode secret data %s", tlsSecretDataKey)
```

Comment:
`failed to decode secret data` -> `failed to decode key data` 

otherwise we can not decide if we have the first error case or this one

Reply:
This should would return `failed to decode secret data tls.key`

Reply:
Updated to make the message clearer: `failed to decode tls.key from secret myapp-secret`